File: csharp/phone-number/PhoneNumber.cs
using System.Text.RegularExpressions;

public class PhoneNumber
{
	private static readonly Regex NumbersOnlyRegex = new Regex(@"[^0-9]");

	public PhoneNumber(string numbers)
	{
		this.Number = PhoneNumber.ParseNumbers(numbers);
		this.AreaCode = this.Number.Substring(0, 3);
	}

	public string Number { get; private set; }

	public string AreaCode { get; private set; }

	private static string ParseNumbers(string numbers)
	{
		var onlyNumbers = PhoneNumber.NumbersOnlyRegex.Replace(numbers, string.Empty);

		if (onlyNumbers.Length == 11 && onlyNumbers[0] == '1')
		{
			onlyNumbers = onlyNumbers.Substring(1);
		}
		else if (onlyNumbers.Length != 10)
		{
			onlyNumbers = "0000000000";
		}

		return onlyNumbers;
	}

	public override string ToString()
	{
		return string.Format("({0}) {1}-{2}", this.AreaCode, this.Number.Substring(3, 3), this.Number.Substring(6));
	}
}

File: javascript/phone-number/phone_number.js
var PhoneNumber = function (n) {
    var cleanNumber = n.replace(/[^\d]/g, "");
    if (cleanNumber.length === 11 && cleanNumber[0] === "1") {
        this._countryCode = cleanNumber[0];
        cleanNumber = cleanNumber.slice(1);
    }

    if (cleanNumber.length === 10) {
        this._number = cleanNumber;
    }
    else {
        this._number = "0000000000";
    }

    this._areaCode = cleanNumber.slice(0, 3);
    this._firstPart = cleanNumber.slice(3, 6);
    this._secondPart = cleanNumber.slice(6);
}

PhoneNumber.prototype.number = function () {
    return this._number;
}

PhoneNumber.prototype.areaCode = function () {
    return this._areaCode;
}

PhoneNumber.prototype.toString = function () {
    return `(${ this._areaCode }) ${ this._firstPart }-${ this._secondPart }`;
}

module.exports = PhoneNumber;